src/nacl/public.py

```diff
@@ -247,3 +247,79 @@ def shared_key(self):
         """
 
         return self._shared_key
+
+
+class SealedBox(encoding.Encodable, StringFixer, object):
+    """
+    The SealedBox class boxes and unboxes messages between an ephemeral
+    key-pair and a long-term public key.
+
+    The ciphertexts generated by :class:`~nacl.public.SecretBox` include
+    the public part of the ephemeral key before the :class:`~nacl.public.Box`
+    ciphertext.
+
+    :param public_key: :class:`~nacl.public.PublicKey` used to encrypt
+        messages and derive nonces
+    :param private_key: :class:`~nacl.public.PrivateKey` used to decrypt
+        messages
+    """
+
+    def __init__(self, recipient_key):
+
+        if isinstance(recipient_key, PublicKey):
+            self._public_key = recipient_key.encode(
+                                                encoder=encoding.RawEncoder)
+            self._private_key = None
+        elif isinstance(recipient_key, PrivateKey):
+            self._private_key = recipient_key.encode(
+                                                encoder=encoding.RawEncoder)
+            self._public_key = recipient_key.public_key.encode(
+                                                encoder=encoding.RawEncoder)
+        else:
+            raise exc.TypeError("SealedBox must be created from "
+                                "a PublicKey or a PrivateKey")
+
+    def __bytes__(self):
+        return self._public_key
+
+    def encrypt(self, plaintext, encoder=encoding.RawEncoder):
+        """
+        Encrypts the plaintext message using a random-generated ephemeral
+        keypair and returns the ciphertext encoded with the encoder.
```

Comment:
Following @maqp suggestion, I've added the note about the public part of the sender's ephemeral keypair being sent along in the following couple of lines. Nonetheless, I'm pushing an attempted rewording in a few minutes.